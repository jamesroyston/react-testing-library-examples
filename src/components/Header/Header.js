import React from 'react';

function Header({text}) {
    return (
        <header>
            <h1 
                data-testid="h1"
                className="fancy-h1"
            >{text}</h1>
        </header>
    )
}

export default Header;