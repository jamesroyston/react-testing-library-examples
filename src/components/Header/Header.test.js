import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Header from './Header';

afterEach(cleanup);

describe.skip("the Header", () => {
    it('renders', () => {
        const { container } = render(<Header text='Hello!' />);
        expect(container).toMatchSnapshot();
    })
    
    it("inserts text in h1", () => {
        const { getByTestId, getByText } = render(<Header text="asdf" />);
        
        expect(getByTestId('h1')).toHaveTextContent('asdf');
        expect(getByText('asdf')).toHaveClass('fancy-h1');
        // toHaveTextContent is a jest-dom assertion, not included in normal jest
        // be mindful of jest-dom vs jest assertions
    })
})