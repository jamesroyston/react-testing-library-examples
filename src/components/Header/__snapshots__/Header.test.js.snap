// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Header renders 1`] = `
<div>
  <header>
    <h1
      class="fancy-h1"
      data-testid="h1"
    >
      Hello!
    </h1>
  </header>
</div>
`;
